ui/static/ui/js/listing.js

```diff
@@ -213,6 +229,72 @@ define(
         });
       });
 
+      var updateCheckDisplay = function(node) {
+        if ($(node).data("selected")) {
+          $(node).find("i").attr("class", "fa fa-check");
+        } else {
+          $(node).find("i").attr("class", "fa fa-arrow-right");
+        }
+      };
+      /**
+       * Update export count badge
+       * @param {number} direction Add this value to
+       * the existing count before rendering
+       */
+      var updateExportCount = function(direction) {
+        var oldCount = $("#export_button").data("export-count");
+        var newCount = oldCount + direction;
+
+        $("#export_button .badge").remove();
+        if (newCount > 0) {
+          var $badge = $("<span class='badge' />");
+          $badge.append(_.escape(newCount));
+          $("#export_button").append($badge);
+        }
+        $("#export_button").data("export-count", newCount);
+      };
+      _.each($(".link-export"), function(node) {
+        updateCheckDisplay(node);
+      });
+      updateExportCount(0);
+
+      // setup click handlers for export links
+      $(".link-export").click(function(event) {
+        event.preventDefault();
+
+        var learningResourceId = $(this).data("learningresource-id");
+        var selected = $(this).data("selected") === true;
+        $(this).data("selected", !selected);
+
+        if (selected) {
+          // remove an item from export cart
+          $.ajax({
+            type: "DELETE",
+            url: "/api/v1/repositories/" + repoSlug +
+            "/learning_resource_exports/" + learningResourceId + "/"
+          }).fail(function() {
+            // TODO: where should this go?
+            console.error("Unable to delete learning resource " +
+              learningResourceId + " from export list");
+          });
+          updateExportCount(-1);
```

Comment:
This line and line 292 below (incrementing and decrementing) the counter look like they always fire, even if there's a failure. Could this cause a UI bug?